packages/hub/services/scheduled-payments/fetcher.ts

```diff
@@ -19,44 +21,74 @@ export default class ScheduledPaymentsFetcherService {
     let prisma = await this.prismaManager.getClient();
     let nowUtc = convertDateToUTC(new Date());
 
-    return prisma.scheduledPayment.findMany({
-      where: {
-        canceledAt: null,
-        creationBlockNumber: {
-          gt: 0,
-        },
-        payAt: {
-          gt: startOfDay(subDays(nowUtc, this.validForDays)),
-          lte: nowUtc,
-        },
-        OR: [
-          {
-            AND: {
-              recurringDayOfMonth: null,
-              scheduledPaymentAttempts: {
-                none: {
-                  OR: [{ status: 'succeeded' }, { status: 'inProgress' }],
-                },
-              },
-            },
-          },
-          {
-            AND: {
-              recurringDayOfMonth: {
-                gt: 0,
-              },
-              recurringUntil: {
-                gte: nowUtc,
-              },
-              scheduledPaymentAttempts: {
-                none: {
-                  status: 'inProgress',
-                },
-              },
-            },
-          },
-        ],
-      },
-    });
+    let results: [{ id: string; started_at: Date }] = await prisma.$queryRaw`SELECT 
+        "scheduled_payments"."id" AS "id",
+        "last_failed_payment_attempts"."started_at"
+      FROM 
+        "scheduled_payments"
+      LEFT JOIN (SELECT "scheduled_payment_attempts"."scheduled_payment_id",
+                  MAX("scheduled_payment_attempts"."started_at") AS "started_at"
+            FROM "scheduled_payment_attempts"
+            WHERE "scheduled_payment_attempts"."status" = 'failed'
+            GROUP BY "scheduled_payment_id") AS "last_failed_payment_attempts"
```

Comment:
Would it be better to say "last_failed_payment_attempt" instead of "last_failed_payment_attempts"? Given it's just one.

Reply:
renamed